src/core/mac/mac.hpp

```diff
@@ -663,6 +676,89 @@ class Mac : public InstanceLocator
      */
     bool IsEnabled(void) const { return mEnabled; }
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    /**
+     * This method sets the CSL accuracy.
+     *
+     * @returns CSL accuracy(ppm).
+     *
+     */
+    uint8_t GetCslAccuracy(void) const { return mSubMac.GetCslAccuracy(); }
+
+    /**
+     * This method sets the CSL accurary.
+     *
+     * @param[in]  aAccuracy  CSL accuracy(ppm).
+     *
+     */
+    void SetCslAccuracy(uint8_t aAccuracy) { mSubMac.SetCslAccuracy(aAccuracy); }
+
+    /**
+     * This method gets the CSL channel.
+     *
+     * @returns CSL channel.
+     *
+     */
+    uint8_t GetCslChannel(void) const { return mSubMac.GetCslChannel(); }
+
+    /**
+     * This method sets the CSL channel.
+     *
+     * @param[in]  aChannel  The CSL channel.
+     *
+     */
+    void SetCslChannel(uint8_t aChannel);
+
+    /**
+     * This method gets the CSL period.
+     *
+     * @returns CSL period in units of 10 symbols.
+     *
+     */
+    uint16_t GetCslPeriod(void) const { return mSubMac.GetCslPeriod(); }
+
+    /**
+     * This method sets the CSL period.
+     *
+     * @param[in]  aPeriod  The CSL period in 10 symbols.
+     *
+     */
+    void SetCslPeriod(uint16_t aPeriod);
+
+    /**
+     * This method gets the CSL timeout.
+     *
+     * @returns CSL timeout in seconds.
+     *
+     */
+    uint32_t GetCslTimeout(void) const { return mSubMac.GetCslTimeout(); }
+
+    /**
+     * This method sets the CSL timeout.
+     *
+     * @param[in]  aTimeout  The CSL timeout in seconds.
+     *
+     * @retval OT_ERROR_NONE           Successfully set the CSL timeout.
+     * @retval OT_ERROR_INVALID_STATE  Invalid state to change this
+     * parameter.
+     *
+     */
+    otError SetCslTimeout(uint32_t aTimeout);
+
+    /**
+     * This method indicates whether the current parent supports CSL.
+     *
+     */
+    bool DoesParentSupportCsl(void) const;
```

Comment:
Remove?
I see the declaration of this method here in `mac.hpp`, but cannot find its implementation and does not seem to be  used anywhere.